src/GNDStk/convert/src/detail.hpp

```diff
@@ -0,0 +1,453 @@
+
+namespace detail {
+
+// -----------------------------------------------------------------------------
+// Helpers for convert(*,JSON)
+// -----------------------------------------------------------------------------
+
+// node2json
+template<
+   template<class...> class METADATA_CONTAINER,
+   template<class...> class CHILDREN_CONTAINER
+>
+bool node2json(
```

Comment:
So this isn't a `detail`. This is the main function of `convert` for a `Node` to `JSON` type. Why not have the contents of this function directly in the `convert` function?

Reply:
The convert() function that's relevant in this discussion is convert(Tree,JSON), i.e. convert Tree (not Node) to JSON. That's the one users should be dealing with. The node-to-json functionality probably wouldn't be of direct interest to a user, and also involves a bit of subtlety (the optional third argument) that allows for child nodes with the same name, under a given parent node, to be handled properly in the JSON, per suggestions in the GNDS manual of how to do so. I intentionally made it a detail function, and with a name other than convert() as we use for user-callable stuff, because I don't think users really need it, and would rather they not mess with it without understanding the subtlety.